pennylane/hf/tapering.py

```diff
@@ -451,16 +461,155 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
                 val[i] *= paulix_sector[idx]
 
     o = []
-    wires_tap = [h.wires[i] for i in range(len(h.wires)) if i not in paulix_wires]
+    wires_tap = [i for i in h.wires if i not in paulix_wires]
     wiremap_tap = dict(zip(wires_tap, range(len(wires_tap) + 1)))
+
     for i in range(len(h.terms[0])):
         s = qml.grouping.pauli_word_to_string(h.terms[1][i], wire_map=wiremap)
         wires = [x for x in h.wires if x not in paulix_wires]
         o.append(
-            qml.grouping.string_to_pauli_word("".join([s[i] for i in wires]), wire_map=wiremap_tap)
+            qml.grouping.string_to_pauli_word(
+                "".join([s[wiremap[i]] for i in wires]), wire_map=wiremap_tap
+            )
         )
 
     c = anp.multiply(val, h.terms[0])
     c = qml.math.stack(c)
 
     return _simplify(qml.Hamiltonian(c, o))
+
+
+def optimal_sector(qubit_op, generators, active_electrons):
```

Comment:
Just a reminder to remove this when this [PR](https://github.com/PennyLaneAI/pennylane/pull/2041) is merged.